mantle/kola/harness.go

```diff
@@ -272,6 +273,74 @@ func testRequiresInternet(test *register.Test) bool {
 	return false
 }
 
+type DenyListObj struct {
+	Pattern string   `yaml:"pattern"`
+	Tracker string   `yaml:"tracker"`
+	Streams []string `yaml:"streams"`
+	Arches  []string `yaml:"arches"`
+}
+
+type ManifestData struct {
+	AddCommitMetadata struct {
+		FcosStream string `yaml:"fedora-coreos.stream"`
+	} `yaml:"add-commit-metadata"`
+}
+
+func parseDenyListYaml() error {
+	var objs []DenyListObj
+
+	// Parse kola-denylist into structs
+	denyListFile, err := ioutil.ReadFile("src/config/kola-denylist.yaml")
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	plog.Debug("Found kola-denylist.yaml. Processing listed denials.")
+	err = yaml.Unmarshal(denyListFile, &objs)
+	if err != nil {
+		return err
+	}
+
+	plog.Debug("Parsed kola-denylist.yaml")
+
+	// Get stream and arch
+	manifestFile, err := ioutil.ReadFile("src/config/manifest.yaml")
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	var manifest ManifestData
+	err = yaml.Unmarshal(manifestFile, &manifest)
+	if err != nil {
+		return err
+	}
+
+	stream := manifest.AddCommitMetadata.FcosStream
+	arch := system.RpmArch()
+
+	// Accumulate patterns filtering by stream and arch
+	plog.Debug("Processing denial patterns from yaml...")
+	for _, obj := range objs {
+		if !hasString(arch, obj.Arches) && len(obj.Arches) > 0 {
+			continue
+		}
+
+		if len(stream) > 0 && !hasString(stream, obj.Streams) && len(obj.Streams) > 0 {
+			continue
+		}
+
+		fmt.Printf("⚠️  Skipping kola test pattern \"%s\":\n", obj.Pattern)
+		fmt.Printf("⚠️	%s\n", obj.Tracker)
```

Comment:
Tab

Reply:
Line 336 has two spaces after the warning character (⚠️) and line 337 has a tab after the warning character. I left those in since I thought they helped improve readability. The two spaces on line 336 prevents the warning character from covering a portion of the output in terminal. The tab on line 337 positions the tracker URL under the corresponding kola test pattern.

Reply:
I somewhat agree with jonathan about the tab. If you want spacing maybe let's just put in spaces (that way it's consistent). 

Alternatively the many ⚠️  characters to mee are a little much (existed in the old implementation too). How about something like:

```

		fmt.Printf("⚠️  Skipping kola test pattern \"%s\":\n", obj.Pattern)
		fmt.Printf("  👉 %s\n", obj.Tracker)
```

So it looks somethink like:

```
⚠️  Skipping kola test pattern "fcos.internet":
  👉 https://github.com/coreos/coreos-assembler/pull/1478
⚠️  Skipping kola test pattern "podman.workflow":
  👉 https://github.com/coreos/coreos-assembler/pull/1478
```

Reply:
That looks good! I'll add that.